### DISCUSSION TITLE
Cannot create SKElement when using SkiaSharp.Views.WPF in Uno on Linux and Net 7.0

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#12039] [Q&A]
I am trying to use SkiaSharp in a WPF project for Linux

I am using .Net 7, VisualStudioCode, Ubuntu 20.04

I have installed SkiaSharp.Views.WPF version 2.88.3.  My MainPage,xaml looks like

<Page
    x:Class="MyUnoApp.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:MyUnoApp"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:skia="clr-namespace:SkiaSharp.Views.WPF;assembly=SkiaSharp.Views.WPF"
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
      <skia:SKElement PaintSurface="OnPaintSurface" IgnorePixelScaling="True" />
    </Grid>
</Page>

When I run dotnet build, i see the following

```
CSC : error UXAML0001: Unable to resolve System.Windows.FrameworkElement (Reason: None, Location:Unknown, Candidates: ) [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage (copy).xaml(12,6): error UXAML0001: An error was found in Grid [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage (copy).xaml(1,2): error UXAML0001: An error was found in Page [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
CSC : error UXAML0001: Processing failed for file /home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage (copy).xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
CSC : error UXAML0001: Unable to resolve System.Windows.FrameworkElement (Reason: None, Location:Unknown, Candidates: ) [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage.xaml(12,6): error UXAML0001: An error was found in Grid [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage.xaml(1,2): error UXAML0001: An error was found in Page [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
CSC : error UXAML0001: Processing failed for file /home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MainPage.xaml.cs(27,18): error CS1061: 'MainPage' does not contain a definition for 'InitializeComponent' and no accessible extension method 'InitializeComponent' accepting a first argument of type 'MainPage' could be found (are you missing a using directive or an assembly reference?) [/home/howard/Desktop/SKElementProblem_B/MyUnoApp/MyUnoApp/MyUnoApp.csproj::TargetFramework=net7.0]
```

SKElement is defined as follows:

```
public class SKElement : System.Windows.FrameworkElement
```

However, System,WIndows.FrameWorkElement is not included in Net 7.0.  Hence, the first error message "Unable to resolve System.Windows.FrameworkElement"

How do I fix this?

Howard Weiss

**Comment by jeromelaban** [comment]
Thanks for the report. 

It seems that you're trying to use SkiaSharp in your application, and you should be taking a look at https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest. In an Uno Platform XAML tree, you need to use Uno Platform compatible controls, such as `SKXamlCanvas`.

**Comment by HowardPWeiss** [comment]
The sample works.  Thank you

I was using SkiaSharp.Views.WPF rather than SkiaSharp.Views.Windows.  From what I found on the Internet,  I thought I needed to use SkiaSharp.Views.WPF and create an SKElement in Uno running on Linux and built using Visual Studio Core.

I still need to modify my application and see if that builds and works, etc.

Howard
